packages/discord.js/src/errors/DJSError.js

```diff
@@ -12,11 +12,15 @@ const { Messages } = require('./Messages.js');
  * @ignore
  */
 function makeDiscordjsError(Base) {
-  return class DiscordjsError extends Base {
+  return class extends Base {
+    static {
+      Object.defineProperty(this, 'name', { value: `Discordjs${Base.name}` });
+    }
+
     constructor(code, ...args) {
       super(message(code, args));
       this.code = code;
-      Error.captureStackTrace?.(this, DiscordjsError);
+      Error.captureStackTrace?.(this, this.constructor);
```

Comment:
I wonder if we can remove this optional chain

Reply:
It looks like both Bun and Deno support this, so I am not sure why this is optional. The blame goes back 8 years, so... let's remove it and see what happens, I'd say.